enterprise/api/v1/views/enterprise_customer.py

```diff
@@ -487,3 +487,32 @@ def unlink_users(self, request, pk=None):  # pylint: disable=unused-argument
                     raise UnlinkUserFromEnterpriseError(msg) from exc
 
         return Response(status=HTTP_200_OK)
+
+    @action(methods=['post'], detail=True, permission_classes=[permissions.IsAuthenticated, IsUseremailInRequest])
+    def unlink_self(self, request, pk=None):  # pylint: disable=unused-argument
+        """
+        Unlink request user from the enterprise.
+        """
+        serializer = serializers.EnterpriseCustomerUnlinkSelfUserSerializer(
+            data=request.data
+        )
+
+        serializer.is_valid(raise_exception=True)
+        enterprise_customer = self.get_object()
+        user_email = serializer.data.get('user_email')
+        is_relinkable = serializer.data.get('is_relinkable', True)
+
+        try:
+            models.EnterpriseCustomerUser.objects.unlink_user(
+                enterprise_customer=enterprise_customer, user_email=user_email, is_relinkable=is_relinkable
+            )
+        except (models.EnterpriseCustomerUser.DoesNotExist, models.PendingEnterpriseCustomerUser.DoesNotExist):
+            msg = "[UNLINK_SELF] User with email {} does not exist in enterprise {}.".format(
+                user_email, enterprise_customer
+            )
+            LOGGER.warning(msg)
```

Comment:
It might be good to respond with an error status here, like `return Response(status=422_UNPROCESSABLE_ENTITY)` and maybe include the exception string, too.